parse.y

```diff
@@ -2613,6 +2614,64 @@ rb_str_to_parser_string(rb_parser_t *p, VALUE str)
     /* Type check */
     return rb_parser_encoding_string_new(p, RSTRING_PTR(str), RSTRING_LEN(str), rb_enc_get(str));
 }
+
+
+static void
+rb_parser_ary_extend(rb_parser_t *p, rb_parser_ary_t *ary, long len)
+{
+    long i;
+    if (ary->capa < len) {
+        ary->capa = len;
+        if (ary->data == NULL) {
+            ary->data = xcalloc(len, sizeof(void *));
+        } else {
+            ary->data = xrealloc(ary->data, sizeof(void *) * len);
+        }
+        for (i = ary->len; i < len; i++) {
+            ary->data[i] = Qnil;
+        }
+    }
+}
+
+static rb_parser_ary_t *
+rb_parser_ary_new_capa(rb_parser_t *p, long len)
+{
+    if (len < 0) {
+        rb_bug("negative array size (or size too big): %ld", len);
+    }
+    rb_parser_ary_t *ary = xcalloc(1, sizeof(rb_parser_ary_t));
+    long i;
+    ary->len = 0;
+    ary->capa = len;
+    if (0 < len) {
+        ary->data = xcalloc(len, sizeof(void *));
+    }
+    else {
+        ary->data = NULL;
+    }
+    for (i = 0; i < len; i++) {
+        ary->data[i] = Qnil;
```

Comment:
Not `Qnil` but `0` (or `NULL`)?

Reply:
Fixed in https://github.com/hasumikin/ruby/commit/9c64383a8a112f2b4c7f9c74bead2e416529160d